src/main/java/org/jitsi/jigasi/JvbConference.java

```diff
@@ -729,6 +724,50 @@ else if (evt.getNewState() == RegistrationState.CONNECTION_FAILED)
         }
     }
 
+    /**
+     * Disco info the addresses, the query is cached and will be returned from cache
+     * once we retrieve it.
+     */
+    private void discoverComponentAddresses()
+    {
+        // we are here in the RegisterThread, and it is safe to query and wait
+        // Uses disco info to discover the AV moderation address.
+        // we need to query the domain part extracted from room jid
+        if (this.callContext.getRoomJidDomain() != null)
+        {
+            try
+            {
+                long startQuery = System.currentTimeMillis();
+
+                // in case when running unittests
+                if (this.getConnection() == null)
+                {
+                    return;
+                }
+
+                DiscoverInfo info = ServiceDiscoveryManager.getInstanceFor(this.getConnection())
+                        .discoverInfo(JidCreate.domainBareFrom(this.callContext.getRoomJidDomain()));
+
+                DiscoverInfo.Identity avIdentity = info.getIdentities().stream().
+                    filter(di -> di.getCategory().equals("component") && di.getType().equals("av_moderation"))
+                        .findFirst().orElse(null);
+
+                if (avIdentity != null && this.getAudioModeration() != null)
+                {
+                    String avModerationAddress = avIdentity.getName();
+                    this.getAudioModeration().setAvModerationAddress(avModerationAddress);
+
+                    logger.info(String.format("%s Discovered %s for %oms.",
```

Comment:
Do we really care about this message only for `avModarationAddress`? I suggest either remove it or move it out of the "if" to be printed right after the `discoverInfo` call

Reply:
I'll remove that and move the print after, I just kept it when moving the stuff.